rust/lance-core/src/datatypes.rs

```diff
@@ -136,11 +136,24 @@ impl TryFrom<&DataType> for LogicalType {
                 DataType::Struct(_) => "large_list.struct".to_string(),
                 _ => "large_list".to_string(),
             },
-            DataType::FixedSizeList(dt, len) => format!(
-                "fixed_size_list:{}:{}",
-                Self::try_from(dt.data_type())?.0,
-                *len
-            ),
+            DataType::FixedSizeList(field, len) => {
+                if field
+                    .metadata()
+                    .get("ARROW:extension:name")
```

Comment:
Do we need to do the same for UUID type as well?

Reply:
Do we want `fixed_size_list<UUID>`? This implementation is mostly a quick hack to get bfloat16. Eventually, I'd like to have proper implementation of FSL fields (with full extension support and nested field support), but I am deferring that work until later as it will be difficult.